sdk/communication/communication-jobrouter/swagger/README.md

```diff
@@ -0,0 +1,160 @@
+
+## Configuration
+
+```yaml
+package-name: azure-communication-jobrouter
+title: JobRouterApiClient
+description: JobRouter Client
+generate-metadata: false
+license-header: MICROSOFT_MIT_NO_VERSION
+output-folder: ../src/generated
+tag: V2022-07-18-preview
+package-version: 1.0.0-beta.1
+model-date-time-as-string: false
+optional-response-headers: true
+typescript: true
+azure-arm: false
+add-credentials: false
+v3: true
+use-extension:
+  "@autorest/typescript": "6.0.0-rc.1"
+input-file: ../swagger/swagger.json
+```
+
+# Remove "x-ms-parameterized-host" section when copying swagger.json
+
+# No empty classes anymore in mapper.ts after assigning ref
+
+# Remove readOnly flag so that child class objectType value can update and pass to endpoint instead of missing
+
+# Change imcompatible TimeSpan type to String which can be parsed by .Net Core TimeSpan system class
+
+# Keep current definition of CommunicationErrorResponse in swagger.json
+
+# Manually add type for repeatabilityHeaders in generated index.ts, should try to find auto solution in future
+
+```yaml
+directive:
+  - from: swagger-document
+    where: "$.definitions.RouterJob.properties.enqueueTimeUtc"
+    transform: >
+      $["x-ms-client-name"] = "enqueuedOn";
+  - from: swagger-document
+    where: "$.definitions.DistributionPolicy.properties.offerTtlSeconds"
+    transform: >
+      $["x-ms-client-name"] = "offerTtlInSeconds";
+  - from: swagger-document
+    where: "$.definitions.JobAssignment.properties.assignTime"
+    transform: >
+      $["x-ms-client-name"] = "assignedOn";
+  - from: swagger-document
+    where: "$.definitions.JobAssignment.properties.closeTime"
+    transform: >
+      $["x-ms-client-name"] = "closedOn";  
+  - from: swagger-document
+    where: "$.definitions.JobAssignment.properties.completeTime"
+    transform: >
+      $["x-ms-client-name"] = "completedOn";
+  - from: swagger-document
+    where: "$.definitions.JobPositionDetails.properties.estimatedWaitTimeMinutes"
+    transform: >
+      $["x-ms-client-name"] = "estimatedWaitTimeInMinutes";
+  - from: swagger-document
+    where: "$.definitions.QueueLengthExceptionTrigger.properties.threshold"
+    transform: >
+      $["x-ms-client-name"] = "maxJobCount";  
+  - from: swagger-document
+    where: "$.definitions.QueueWeightedAllocation.properties.weight"
+    transform: >
+      $["x-ms-client-name"] = "weightTotalAsOne";     
+      
+  - from: swagger-document
+    where: "$.definitions.DistributionPolicy.properties.mode"
+    transform: >
+      $["$ref"] = "#/definitions/DistributionMode";
+
+  - from: swagger-document
+    where: "$.definitions.PagedDistributionPolicy.properties.mode"
+    transform: >
+      $["$ref"] = "#/definitions/DistributionMode";
+      
+  - from: swagger-document
+    where: "$.definitions.CompositeModeIteration.properties.mode"
+    transform: >
+      $["$ref"] = "#/definitions/DistributionMode";
+
+  - from: swagger-document
```

Comment:
There seems to be heavy transforms to the swagger. Have you considered changing the swagger instead? I assume most of these transformations are also done for the other languages